nixos/modules/config/users-groups.nix

```diff
@@ -406,6 +414,12 @@ in {
   imports = [
     (mkAliasOptionModule [ "users" "extraUsers" ] [ "users" "users" ])
     (mkAliasOptionModule [ "users" "extraGroups" ] [ "users" "groups" ])
+    (mkChangedOptionModule
+      [ "security" "initialRootPassword" ]
+      [ "users" "users" "root" "initialHashedPassword" ]
+      (cfg: if cfg.security.initialHashedPassword == "!"
+            then null
+            else cfg.security.initialHashedPassword))
```

Comment:
This means that the change is backwards-compatible with people setting `security.initialRootPassword`, right?

Reply:
Yes, it is. Also this option has been hidden since 2014, so the impact should be small.